master/internal/rm/kubernetesrm/spec.go

```diff
@@ -349,45 +359,45 @@ func (p *pod) createPriorityClass(name string, priority int32) error {
 	return err
 }
 
-func (p *pod) configurePodSpec(
+func (j *job) configurePodSpec(
 	volumes []k8sV1.Volume,
 	determinedInitContainers k8sV1.Container,
 	determinedContainer k8sV1.Container,
 	sidecarContainers []k8sV1.Container,
 	podSpec *k8sV1.Pod,
 	scheduler string,
-) *k8sV1.Pod {
+) *batchv1.Job {
 	if podSpec == nil {
 		podSpec = &k8sV1.Pod{}
 	} else {
 		podSpec = podSpec.DeepCopy()
 	}
```

Comment:
this doesn't make sense -- returning &k8sv1.Pod{} with return type Job?

Reply:
it doesn't get returned if you look down at the very bottom. jobs have an embedded pod spec so most of this just forms a pod spec unchanged.